driver/main.c

```diff
@@ -669,9 +678,11 @@ static int force_tp_set(struct ppm_consumer_t *consumer, u32 new_tp_set)
 			continue;
 		}
 
-		switch(idx)
+		// Kmod keeps a 0-indexed mask of tracepoints;
+		// we want to find proper PPM_SC codes here
+		switch(idx + PPM_SC_TP_START)
```

Comment:
In kmod we use tracepoints shifted to `0 - Ntracepoints`, but enums start from `PPM_SC_TP_START`.